File: backend/src/main/java/com/leavemarker/entity/LeavePolicy.java
package com.leavemarker.entity;

import com.leavemarker.enums.LeaveType;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "leave_policies",
       uniqueConstraints = {
           @UniqueConstraint(columnNames = {"company_id", "leave_type"})
       })
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class LeavePolicy extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_id", nullable = false)
    private Company company;

    @Enumerated(EnumType.STRING)
    @Column(name = "leave_type", nullable = false, length = 30)
    private LeaveType leaveType;

    @Column(nullable = false)
    private Integer annualQuota;

    @Column(nullable = false)
    @Builder.Default
    private Double monthlyAccrual = 0.0;

    @Column(nullable = false)
    @Builder.Default
    private Boolean carryForward = false;

    @Column(nullable = false)
    @Builder.Default
    private Integer maxCarryForward = 0;

    @Column(nullable = false)
    @Builder.Default
    private Boolean encashmentAllowed = false;

    @Column(nullable = false)
    @Builder.Default
    private Boolean halfDayAllowed = true;

    @Column(nullable = false)
    @Builder.Default
    private Boolean active = true;
}

File: backend-go/internal/models/leave_policy.go
package models

import "github.com/leavemarker/backend-go/internal/enums"

// LeavePolicy represents a leave policy entity
type LeavePolicy struct {
	BaseModel
	CompanyID          uint            `gorm:"not null;index;uniqueIndex:idx_company_leave_type" json:"companyId"`
	LeaveType          enums.LeaveType `gorm:"size:30;not null;uniqueIndex:idx_company_leave_type" json:"leaveType"`
	AnnualQuota        int             `gorm:"not null" json:"annualQuota"`
	MonthlyAccrual     float64         `gorm:"default:0.0" json:"monthlyAccrual"`
	CarryForward       bool            `gorm:"default:false" json:"carryForward"`
	MaxCarryForward    int             `gorm:"default:0" json:"maxCarryForward"`
	EncashmentAllowed  bool            `gorm:"default:false" json:"encashmentAllowed"`
	HalfDayAllowed     bool            `gorm:"default:true" json:"halfDayAllowed"`
	Active             bool            `gorm:"default:true" json:"active"`

	// Relationships
	Company Company `gorm:"foreignKey:CompanyID" json:"-"`
}

func (LeavePolicy) TableName() string {
	return "leave_policies"
}
